src/NHibernate/Type/ClassMetaType.cs

```diff
@@ -16,7 +16,8 @@ public partial class ClassMetaType : AbstractType
 	{
 		public override SqlType[] SqlTypes(IMapping mapping)
 		{
-			return new SqlType[] { NHibernateUtil.String.SqlType };
+			var types = new[] { NHibernateUtil.String.SqlType };
+			return mapping?.Dialect.RefineSqlTypes(this, types) ?? types;
```

Comment:
I have added that logic for each type, excepted those yielding no types and those building them from other types.